### ISSUE TITLE
Error context is lost when caught by effection

### ISSUE THREAD

**Issue by lc-mm** [thefrontside/effection#800]
Running v2.0.7 there error object returned only includes the name, message, and effectionTrace

The issue is that the error being thrown in my case is a custom object that does not include a name or a message. 

Can the code be modifiued to also include the raw error object?

```diff
  return Object.create(error, {
    name: {
      get: () => error.name,
      enumerable: true,
    },
    message: {
      get: () => error.message,
      enumerable: true,
    },
+    error: {
+      get: () => error,
+      enumerable: true,
+    },
    effectionTrace: {
      value: [...(error.effectionTrace || []), info],
      enumerable: true,
    }
  });
```

**Comment by cowboyd** [comment]
@lc-mm Absolutely. I think that "cause" is the canonical name for it https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause 

Would you be able to submit a PR for it?

You may also consider using Effective v3 (currently in beta) which does not do any special error handling.



<blockquote><img src="https://developer.mozilla.org/mdn-social-share.cd6c4a5a.png" width="48" align="right"><div><strong><a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause">Error: cause - JavaScript | MDN</a></strong></div><div>The cause data property of an Error instance indicates the specific original cause of the error.</div></blockquote>

**Comment by lc-mm** [comment]
Thanks @cowboyd here is a PR #801

**Comment by cowboyd** [comment]
This has been released with Effection 2.0.8 https://github.com/thefrontside/effection/pull/802
